common/src/main/java/com/gregtechceu/gtceu/api/data/worldgen/generator/VeinedVeinGenerator.java

```diff
@@ -75,30 +77,34 @@ public class VeinedVeinGenerator extends VeinGenerator {
     @Setter
     public float maxRichnessThreshold = 0.6f;
     @Setter
-    public float denseBlockChance = 0.02f;
-    //@Setter
-    //public float oreGapNoiseSkipThreshold = -0.3f;
+    public float rareBlockChance = 0.02f;
 
-    public VeinedVeinGenerator(GTOreFeatureEntry entry) {
+    public VeinedVeinGenerator(GTOreDefinition entry) {
         super(entry);
     }
 
     @Override
     public Map<Either<BlockState, Material>, Integer> getAllEntries() {
-        return Map.of(
-                oreBlock.map(state ->
-                                state.stream().map(target -> Either.<BlockState, Material>left(target.state)),
-                        material -> Stream.of(Either.<BlockState, Material>right(material))), 1,
-                denseBlock.map(state ->
-                                state.stream().map(target -> Either.<BlockState, Material>left(target.state)),
-                        material -> Stream.of(Either.<BlockState, Material>right(material))), 1,
-                Stream.of(Either.<BlockState, Material>left(fillerBlock)), 1
-        ).entrySet().stream().flatMap(entry -> entry.getKey().map(either -> Map.entry(either, entry.getValue()))).collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+        var s1 = this.oreBlocks.stream()
+                .flatMap(definition ->
+                        definition.block.map(state ->
+                                        state.stream().map(target -> Map.entry(Either.<BlockState, Material>left(target.state), definition.weight)),
+                                material -> Stream.of(Map.entry(Either.<BlockState, Material>right(material), definition.weight))));
+        var s2 = this.rareBlocks.stream()
+                .flatMap(definition ->
+                        definition.block.map(state ->
+                                        state.stream().map(target -> Map.entry(Either.<BlockState, Material>left(target.state), definition.weight)),
+                                material -> Stream.of(Map.entry(Either.<BlockState, Material>right(material), definition.weight))));
```

Comment:
Just a formatting suggestion for better readability ^^

```suggestion
        var s1 = this.oreBlocks.stream().flatMap(definition ->
                definition.block.map(
                        state -> state.stream().map(target -> Map.entry(Either.<BlockState, Material>left(target.state), definition.weight)),
                        material -> Stream.of(Map.entry(Either.<BlockState, Material>right(material), definition.weight))
                )
        );
        var s2 = this.rareBlocks.stream().flatMap(definition ->
                definition.block.map(
                        state -> state.stream().map(target -> Map.entry(Either.<BlockState, Material>left(target.state), definition.weight)),
                        material -> Stream.of(Map.entry(Either.<BlockState, Material>right(material), definition.weight))
                )
        );
```